duckpgq/src/duckpgq/functions/scalar/iterativelength.cpp

```diff
@@ -110,11 +131,13 @@ static void IterativeLengthFunction(DataChunk &args, ExpressionState &state,
         int64_t dst_pos = vdata_dst.sel->get_index(search_num);
         if (!vdata_src.validity.RowIsValid(src_pos)) {
           result_validity.SetInvalid(search_num);
-          result_data[search_num] = (uint64_t)-1; /* no path */
+          result_data[search_num] = (int64_t)-1; /* no path */
         } else if (src_data[src_pos] == dst_data[dst_pos]) {
           result_data[search_num] =
```

Comment:
I think this is the problematic part I mentioned on Discord
If the source == destination, but the lower bound > 0, we still want to search for a path that is within the bounds.